### PR TITLE
Add HTTPS support to bucket server

### PR DESCRIPTION
**Pull request by makkes**
The bucket server used by `gitops beta run` now serves over HTTP as well as HTTPS. HTTP is still necessary as Flux's Bucket resource doesn't have a field for providing a custom CA certificate.

This is a backwards-incompatible change as the ports the server is listening on have to provided through flags. Also, providing TLS cert and key is mandatory.

### CODE REVIEW

**Comment by chanwit** [comment]
@makkes `make lint` would help you get rid of those errors.

File: pkg/http/server_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package http_test
```
Comment (chanwit) [inline pkg/http/server_test.go:1]: I think we don't need a separate package for test cases, right?

File: cmd/gitops-bucket-server/main.go [hunk 2]
```diff
@@ -2,60 +2,59 @@ package main
...
+		certFile, keyFile   string
+	)
 
-	// create a listener with the desired port.
-	listener, err := net.Listen("tcp", ":"+port)
-	if err != nil {
-		log.Fatal(err)
-	}
+	flag.IntVar(&httpPort, "http-port", 9000, "TCP port to listen on for HTTP connections")
+	flag.IntVar(&httpsPort, "https-port", 9001, "TCP port to listen on for HTTPS connections")
```
Comment (chanwit) [inline cmd/gitops-bucket-server/main.go:36]: The default HTTPS port would be another number like 9443, as we are using `9001` for our default dashboard port.

**Review by chanwit — CHANGES_REQUESTED** [review]

File: pkg/http/server_test.go (hunk 1 above)
Comment (makkes) [reply pkg/http/server_test.go:1]: I usually do that to make sure to only test the public surface of a package.

**Review by chanwit — APPROVED** [review]

**Pull request merged**